hibernate-core/src/main/java/org/hibernate/dialect/DB2Dialect.java

```diff
@@ -572,6 +572,8 @@ protected SqlTypeDescriptor getSqlTypeDescriptorOverride(int sqlCode) {
 			// Therefore here we overwrite the sql type descriptors to
 			// use the non-N variants which are supported.
 			switch ( sqlCode ) {
+				case Types.BOOLEAN:
+					return SmallIntTypeDescriptor.INSTANCE;
```

Comment:
Why precisely was this needed?

Reply:
DB2 has no booleans. I adapted this from https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/dialect/DB2Dialect.java#L502 as I thought this was missed.

Reply:
> DB2 has no booleans.

I'm pretty sure that can't be right, since I know for sure I extensively tested the new `DB2Dialect`, and I'm using `boolean` as the mapping for both `BIT` and `BOOLEAN`.

Reply:
Apparently the answer is that `BOOLEAN` was introduced in DB2 11. So it looks like we need to make that type mapping conditional on the database version.